### PR TITLE
Remove Logger.cc

### PR DESCRIPTION
**Pull request by momoko-h**
One of the two overloads in this file is not used any more since the removal
of the old logging functions and the other one just passes its arguments on
to Logger_log.

Btw, is it really not possible to get the current log level from C++? I tried profiling why an opponent's turn takes so long and a significant amount of time is spent just formatting all those debug messages even when they are not even printed, so it would be great if `LogMessageST` could check the log level before it calls ST::format and Logger_log.

### CODE REVIEW

**Comment by selaux** [comment]
Thanks. Getting the log level from rust is currently not implemented. Here is a commit that does it: https://github.com/ja2-stracciatella/ja2-stracciatella/commit/39528ce39f9b81e98570d2bcde3cce7cf567bf60. Do you want to include it in this PR?

**Comment by momoko-h** [comment]
Yes, I think these two issues are closely related, so we might as well combine them.

**Comment by selaux** [comment]
Is the performance difference significant now? Would you mind sharing a flamegraph or something?

File: rust/stracciatella/src/logger.rs [hunk 1]
```diff
@@ -165,7 +181,12 @@ impl Logger {
 
     /// Sets the global log level to a specific value
     pub fn set_level(level: LogLevel) {
-        GLOBAL_LOG_LEVEL.store(level.into(), Ordering::Relaxed);
+        RuntimeLevelFilter::set_global_log_level(level.into())
+    }
+
+    /// Sets the global log level to a specific value
```
Comment (selaux) [inline rust/stracciatella/src/logger.rs:187]: I know this came from me, but there seems to be a copy-paste-error in the comment...

File: rust/stracciatella_c_api/src/c/logger.rs [hunk 2]
```diff
@@ -28,6 +28,12 @@ pub extern "C" fn Logger_setLevel(level: LogLevel) {
     Logger::set_level(level)
 }
 
+/// Set log level
```
Comment (selaux) [inline rust/stracciatella_c_api/src/c/logger.rs:31]: I know this came from me, but there seems to be a copy-paste-error in the comment

**Comment by ja2-derek** [comment]
> Is the performance difference significant now? Would you mind sharing a flamegraph or something?

I also want to know how things have changed since 0.17.. The last time we did this was:  https://github.com/ja2-stracciatella/ja2-stracciatella/issues/1136

**Comment by momoko-h** [comment]
This is what current master looks like:
KCG_Master

`CalcChanceToGetThrough` spends more than half of its time logging.

This is the same turn, as much as the RNG allows, with this patch applied:
KCG_RemoveLogger

Even though the number of LogFormatST is even higher, `CalcChanceToGetThrough` doesn't even make it to the top of the list.

**Comment by ja2-derek** [comment]
The call graph indeed looks much better. Thanks!

**Comment by selaux** [comment]
Looks great. Thanks a lot!

**Pull request merged**